### ISSUE TITLE
[Feature] Sun protection item - Umbrella

### ISSUE THREAD

**Issue by lusifer173** [TeamLapen/Vampirism#542]
I would like if you added umbrellas or something that allowed us to go out during the day

I think it does considering the fact that the hunter faction has garlic which disables our flight , it also has obsidian armor which is way better then the armor we have , it also has rapid fire crossbows which allow them to finish a fight without even trying , and even if they don't use those things if you fight in the open when the day comes you nearly instantly lose the fight. It doesn't even have to make the vampire faction "op" it doesn't have to "take away their only weakness"  for multiple reasons. For starters we have many weakness being garlic, potions, vampire slayer enchantments, and just for insult to injury hunters even have armor that is way better then the armor we can have. Even if this does make vampires "to op" you could design it to where is makes vampires unable to attack while using or holding an umbrella. Further more although vampires can fly it can be disabled, even though we have the rage ability it doesn't last long and has a long cool down, even though we can teleport it doesn't help us in fights as if we get close to a hunter they have vampire slayer and if we try and run they have rapid fire crossbows that might i add can give the garlic effect and even place fire, even though we have the freeze ability to use it well we have to get close which as i have already said puts us in range of the vampire slayer enchantment. I'm simply asking for a way to play minecraft and not be forced to sit underneath a tree for ten minutes and be forced to watch others play.

Sadly i cannot provide any help and or textures for this item.

**Comment by lunofe** [comment]
Sunscreens, whether an item, ability or beacons, were suggested many times. It has never been added, because it would break the balance between vampires and hunters. Vampires already have powerful abilities which allow them to have advantages in many situations, for example the bat mode, freeze, vampire rage, regeneration and sunscreen. The only downside of being a vampire, is being a vampire, which means, that you will burn in the sun. Everything else is giving you advantages. So, if we remove that weakness, it would totally break the balance. Yes, even if you can't do combat while using it: Simply switch the item/weapon very fast and you won't even get the nausea effect. There is already the learnable skill to survive some moments in the sun and I doubt and hope, that it'll never be more than that. Vampires without being burnt by the sun, aren't Vampires.

**Comment by lunofe** [comment]
> I'm simply asking for a way to play minecraft and not be forced to sit underneath a tree for ten minutes

The life of a vampire.

**Comment by PendragonII** [comment]
> I would like if you added umbrellas or something that allowed us to go out during the day

There is a skill for permanently reducing the intake damage, and another skill for temporarily nullifying the the Sun's damage. The Sun is also both lore and an ingame balance mechanic.

### Below is a rebute of the statements of the lack of balance. 

**TL;DR: Vampires have many advantages, adding Umbrellas will skew the few disadvantageous of the Vampire therefore breaking the Hunter-Human-Vampire balance:**

> I think it does considering the fact that the hunter faction has garlic which disables our flight [...] Further more although vampires can fly it can be disabled

Vampire as bats have high travel speed,  can conduct renaissance, circumvent protective barriers, et cetera; Garlic Diffusers balance this game mechanic by disabling Vampire's Bat Skill for _only_ a selected region.

Garlic Diffusers are also late-game-expensive, requiring late-level Holy Water skill, Vampire Repulsion skill, diamonds and obsidian.

> it also has obsidian armor which is way better then the armor we have [...] and just for insult to injury hunters even have armor that is way better then the armor we can have.

As of writing, the obsidian armor requires multiple skill to craft, late-game-level expensive and gives a slowness effect. Wearing all pieces of the obsidian armor amounts to 25% speed reduction; comparative to the Vampire whom has passive speed. _This is balanced_ when also looking at the benefits of obsidian armor.

>  rapid fire crossbows which allow them to finish a fight without even trying [...] if [vampires] try and run they have rapid fire crossbows

This is assuming the opponent has no option but there is many, including vampires: Shields to block the bolts, cover, retaliation with bows & arrows, blood projectiles (Vampire skill)

> if [vampires] fight in the open when the day comes you nearly instantly lose the fight

There's a biome called Vampire Forest and Vampire Villages where no Sunlight can have effect.
Both these places also have higher spawn rate for Vampire NPC's adding to a kin-advantage against Hunters and Humans.

_Vampire Villages can be made by moving or curing villagers, and placing a Vampirism-Mod Totem, then the surrounding land will become Vampire Forest-like._

> we have many weakness being garlic, potions, vampire slayer enchantments, 

These weakness are applicable through the Hunter Players[1], read below.
**[1]** As of writing, Hunter NPC's do not implement garlic, potions, vampire slayer enchantments into their combat. Read below.

> Even if this does make vampires "to op" you could design it to where is makes vampires unable to attack while using or holding an umbrella.

While the Hunter chases? Unfortunately, Hunters lack the Vampiric's speed _(beneficial for Vampires!)_.
Alternatively, Vampires could _(reasonably so)_ make a roof or go to Vampire Forests and Vampire Villages which also keeps the Vampire's skill intact.

> even though we have the rage ability it doesn't last long and has a long cool down

A minute in combat is quite some time for the focused. The Rage ability is _useful_ when used _strategically_ in combat (among other things). The long cooldown is a balance mechanic so the Rage ability cannot be abused nor overpowering.

> even though we can teleport it doesn't help us in fights as if we get close to a hunter they have vampire slayer and [...] that might i add can give the garlic effect and even place fire, even though we have the freeze ability to use it well we have to get close which as i have already said puts us in range of the vampire slayer enchantment.

Teleportation, freeze, are both combat skills: it takes ingame practice to use it effectively.
Garlic and fire are both avoidable: bring a water bucket and should a Hunter bring a Garlic Diffuser it is _(very much so)_ within the Vampire's power to run out of the Garlic Diffuser's region-of-effect (Vampires have passive speed, accelerated by a dedicated speed boost skill).

This is balanced.

> I'm simply asking for a way to play minecraft and not be forced to sit underneath a tree for ten minutes and be forced to watch others play.

I play on the Vampirism Server, Vampire Players create makeshift roofs for traveling with their Hunter friends (you read that correctly), or live in Vampire Forests or Vampire Villages.

In singleplayer or if you (or a friend) create a server the Sun's damage is configurable.

--

Here is my speculative opinion on Hunters and Vampires:
 - Vampires are generalized roleplay characters: Turning to bats to fly, passive speed with each level increase, ignored by fellow _vampires_ and _undead zombies_ passively and _creepers_ additionally with a skill, teleportation, strength from rage. Indeed, Vampires are generalized and well fitted for many roleplayers.
 - Hunters are specialized roleplay characters: Hunters specialize in (unsurprisingly) hunting undead such as vampires. Holy Water, Crossbows, Garlic, all these things are tailored to hunting Vampires. This roleplay style is combat focused.

Here is the equilibrium. Vampires suffer from the Sun as part of the game balancing mechanics.
The license allows you to copy and modify the code however you like. You may do so yourself or hire a programmer to do so.

### Most of these things, if not all, are configurable in Vampirism's mod settings.

**Comment by maxanier** [comment]
It is difficult to balance the mod for both single player and multiplayer gameplay as well as heavily modded and pure Vampirism. Therefore, Vampirism is highly configurable as Pendragonll pointed out. You can e.g. adjust the rage cooldown as you wish.

Regarding the sun protection. I might add a disabled by default umbrella item, which takes the place of your main weapon and slows you down. Doesn't hurt anyone and shouldn't take to long. But let's see

**Comment by wolf653** [comment]
> There's a biome called Vampire Forest and Vampire Villages where no Sunlight can have effect.
> Both these places also have higher spawn rate for Vampire NPC's adding to a kin-advantage against Hunters and Humans.

Doesn't this just help hunters level up?
